src/flow/utils.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { typeAnnotation } from "babel-types";
+import invariant from "../invariant.js";
+
+export function flowAnnotationToObject(annotation: typeAnnotation) {
```

Comment:
It's weird that this doesn't return an object for some objects. `flowAnnotationToObjectTypeTemplate`?